crates/hir_ty/src/tests/regression.rs

```diff
@@ -832,7 +832,7 @@ fn issue_4966() {
             365..390 'Repeat...nner }': Repeat<Map<|&f64| -> f64>>
             383..388 'inner': Map<|&f64| -> f64>
             401..404 'vec': Vec<IntoIterator::Item<Repeat<Map<|&f64| -> f64>>>>
-            407..416 'from_iter': fn from_iter<IntoIterator::Item<Repeat<Map<|&f64| -> f64>>>, Repeat<Map<|&f64| -> f64>>>(Repeat<Map<|&f64| -> f64>>) -> Vec<IntoIterator::Item<Repeat<Map<|&f64| -> f64>>>>
+            407..416 'from_iter': fn from_iter<IntoIterator::Item<Repeat<Map<|&f64| -> f64>>>, Repeat<Map<|&f64| -> f64>>>(Repeat<Map<|&f64| -> f64>>) -> Vec<<Repeat<Map<|&f64| -> f64>> as IntoIterator>::Item>
```

Comment:
this change is surprising. are we still calling `display` somewhere in `check_infer`?